src/Resources/Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -209,14 +222,35 @@ public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions
                     }
 
                     principalId = adObject.Id.ToString();
-                    odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AssignedTo(principalId));
+                    if (!hasScope)
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AssignedTo(principalId));
+                    }
+                    else
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => (f.AtScope() && f.AssignedTo(principalId)));
+                    }
                 }
                 else
                 {
                     principalId = string.IsNullOrEmpty(options.ADObjectFilter.Id) ? adObject.Id.ToString() : options.ADObjectFilter.Id;
-                    odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
+                    if (!hasScope)
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
+                    } 
+                    else
+                    {
+                        // atScope() can't use with pricipalId eq {objectId} expression, Principal id could be group id.
+                        // Only pricipalId eq {objectId} can list role assignments for Group.
```

Comment:
this might be poorly phrased, the way it's worded it looks like it contradicts your comment just one line before

Reply:
<html>
<body>
<!--StartFragment-->

$filter=principalId+eq+'{objectId}' | Lists role assignments for a specified user, group, or service principal.
-- | --


<!--EndFragment-->
</body>
</html>
if you use atScope() and principalId+eq+'{objectId}' together, it will throw exception because the API doesn't allow it. 
objectId could represent a group, so can't use atScope() and assignedTo('{objectId}').